lib/datadog/search-syntax/src/vrl.rs

```diff
@@ -0,0 +1,549 @@
+                (ComparisonValue::Unbounded, ComparisonValue::Unbounded) => {
+                    make_function_call("exists", vec![make_query(attr)])
+                }
+                // Unbounded lower. Wrapped in a container group for negation compatibility.
+                (ComparisonValue::Unbounded, _) => make_container_group(make_op(
+                    make_node(make_query(attr)),
+                    if upper_inclusive {
+                        ast::Opcode::Le
+                    } else {
+                        ast::Opcode::Lt
+                    },
+                    make_node(ast::Expr::Literal(make_node(upper.into()))),
+                )),
+                // Unbounded upper. Wrapped in a container group for negation compatibility.
+                (_, ComparisonValue::Unbounded) => make_container_group(make_op(
+                    make_node(make_query(attr)),
+                    if lower_inclusive {
+                        ast::Opcode::Ge
+                    } else {
+                        ast::Opcode::Gt
+                    },
+                    make_node(ast::Expr::Literal(make_node(lower.into()))),
+                )),
+                // Definitive range.
+                _ => make_container_group(make_op(
+                    make_node(make_op(
+                        make_node(make_query(attr.clone())),
+                        if lower_inclusive {
+                            ast::Opcode::Ge
+                        } else {
+                            ast::Opcode::Gt
+                        },
+                        make_node(ast::Expr::Literal(make_node(lower.into()))),
+                    )),
+                    ast::Opcode::And,
+                    make_node(make_op(
+                        make_node(make_query(attr)),
+                        if upper_inclusive {
+                            ast::Opcode::Le
+                        } else {
+                            ast::Opcode::Lt
+                        },
+                        make_node(ast::Expr::Literal(make_node(upper.into()))),
+                    )),
+                )),
+            },
+            // Negation.
+            QueryNode::NegatedNode { node } => make_not(ast::Expr::from(*node)),
+            // Compound.
+            QueryNode::Boolean { oper, nodes } => {
+                make_container_group(nest_exprs(nodes.into_iter(), oper))
+            }
+        }
+    }
+}
+
+/// Creates a VRL node with a default span.
+fn make_node<T>(node: T) -> ast::Node<T> {
+    ast::Node::new(Span::default(), node)
+}
+
+/// Converts a field/facet name to the VRL equivalent. Datadog payloads have a `message` field
+/// (which is used whenever the default field is encountered. Facets are hosted on .custom.*.
+fn normalize_tag<T: AsRef<str>>(value: T) -> String {
+    let value = value.as_ref();
+    if value.eq(grammar::DEFAULT_FIELD) {
+        return "message".to_string();
+    }
+
```

Comment:
Another thing we need to check here is whether the field(which doesn't start with `@`) belongs to the list of the root/reserved fields(service, hostname, source, message, timestamp etc). If not - we need to look up for this field in `tags`.

Reply:
Thanks -- do you know if there's a canonical list of reserved fields that I can use as a reference please?

Do these change per account -- or are they statically defined by Datadog?

Reply:
They are statically defined https://docs.datadoghq.com/logs/processing/attributes_naming_convention/#reserved-attributes 
plus implicitely it also includes `timestamp` and `tags`. 
Technically for JSON logs you could define which attributes map to any of these fields - this is called JSON pre-processing, but as far as I remember we decided not to support json pre-processing for now.

Reply:
Thanks, that's good to know. I'll make sure those fields are distinguished.